community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -922,4 +895,69 @@ public GraphPropertiesProxy newGraphPropertiesProxy()
     {
         return new GraphPropertiesProxy( this );
     }
+
+    private final class NodeCursorResourceIterator implements ResourceIterator<Node>
+    {
+        private static final long UNINITIALIZED = -2L;
+        private static final long NO_ID = -1L;
+        private final NodeLabelIndexCursor cursor;
+        private final Statement statement;
+        private long next;
+        private boolean closed;
+
+        NodeCursorResourceIterator( NodeLabelIndexCursor cursor, Statement statement )
+        {
+            this.cursor = cursor;
+            this.statement = statement;
+            next = -2L;
```

Comment:
Should be UNINITIALIZED

Reply:
Or perhaps the first fetchNext could be called in the constructor to remove the need for this special value?